source/include/sigv4_internal.h

```diff
@@ -113,10 +113,15 @@
  * @brief A helper macro to print insufficient memory errors.
  */
 #define LOG_INSUFFICIENT_MEMORY_ERROR( purposeOfWrite, bytesExceeded )                       \
-    do {                                                                                     \
+    {                                                                                        \
         LogError( ( "Insufficient memory provided to " purposeOfWrite ", bytesExceeded=%lu", \
                     ( unsigned long ) ( bytesExceeded ) ) );                                 \
-    } while( 0 )
+    }
```

Comment:
This was removed due to a MISRA warning that `0` is not a boolean, but if we want to keep the `do-while`, then it can be a constant condition:
```suggestion
    } while( 1 == 0 )
```

Reply:
Should we keep the do-while? It's a little unconventional that just using what we have now.